### PR TITLE
feat(j-s): Defender access changes

### PR DESCRIPTION
**Pull request by unakb**
[Aðgengi allra lögmanna (skráðir í lögmannaskrá) ](https://app.asana.com/0/1199153462262248/1205581325882016)

## What

Check lawyer registry when defenders without cases attempt to log in 

## Why

So the log in experience is better for defenders with no active cases

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/api/src/app/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -162,6 +161,7 @@ export class AuthController {
 
       if (verifiedUserToken) {
         this.logger.debug('Token verification successful')
+        res.clearCookie(CODE_VERIFIER_COOKIE.name, CODE_VERIFIER_COOKIE.options)
```
Comment (gudjong) [inline apps/judicial-system/api/src/app/modules/auth/auth.controller.ts:164]: 🧹

File: apps/judicial-system/api/src/app/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -1,17 +1,20 @@
 import fetch from 'isomorphic-fetch'
 import jwt from 'jsonwebtoken'
 import jwksClient from 'jwks-rsa'
+import { uuid } from 'uuidv4'
 
 import { Inject, Injectable, UnauthorizedException } from '@nestjs/common'
 import { ConfigType } from '@nestjs/config'
 
-import type { User } from '@island.is/judicial-system/types'
+import { type User, UserRole } from '@island.is/judicial-system/types'
```
Comment (gudjong) [inline apps/judicial-system/api/src/app/modules/auth/auth.service.ts:9]: Interesting, didn't know this was possible.

File: apps/judicial-system/api/src/app/modules/auth/auth.service.ts [hunk 3]
```diff
@@ -32,18 +35,50 @@ export class AuthService {
...
+        `${this.config.backendUrl}/api/cases/limitedAccess/defender?nationalId=${nationalId}`,
+        {
+          headers: { authorization: `Bearer ${this.config.secretToken}` },
+        },
+      )
+      if (res.ok) {
+        return await res.json()
+      }
+    } catch (error) {
+      console.warn('Defender not found', error)
```
Comment (gudjong) [inline apps/judicial-system/api/src/app/modules/auth/auth.service.ts:49]: Please check specifically for a NotFoundException and only log and pass through for that exceptions. Also, consider logging as info. An error is logged futher down the call path "Blocking login attempt from an unauthorized user". That message should probably be logged as a warning instead.

If the exception is anything else it should be rethrown as it indicates an error on the backend or a problem with the connection to the backend.

File: apps/judicial-system/api/src/app/modules/auth/auth.service.ts [hunk 4]
```diff
@@ -32,18 +35,50 @@ export class AuthService {
...
+          mobileNumber: lawyerRegistryInfo.phoneNr,
+          active: true,
+          title: 'verjandi',
+          id: uuid(),
+          created: new Date().toString(),
+          modified: new Date().toString(),
+        } as User
+      }
+    } catch (error) {
+      console.warn('Defender not found in lawyer registry', error)
```
Comment (gudjong) [inline apps/judicial-system/api/src/app/modules/auth/auth.service.ts:78]: Consider logging as info for the same reason as above. Also, consider a more accurate message as the reason for the failure could be anything.

File: apps/judicial-system/api/src/app/modules/defender/defender.service.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    })
+
+    if (response.ok) {
+      const lawyers = await response.json()
+      const lawyersMapped = (lawyers || []).map(mapToLawyer)
+      return lawyersMapped
+    }
+
+    const reason = await response.text()
+    console.error('Failed to get lawyers:', reason)
```
Comment (gudjong) [inline apps/judicial-system/api/src/app/modules/defender/defender.service.ts:32]: Please remove console.log. If you still want to log, consider injecting a logger and use it instead. Also, please do not log as error as that will trigger alerts in operation.

File: apps/judicial-system/api/src/app/modules/auth/auth.service.ts (hunk 3 above)
Comment (gudjong) [reply apps/judicial-system/api/src/app/modules/auth/auth.service.ts:49]: Please do not use console logging. Instead, inject a logger and use it instead.

File: apps/judicial-system/api/src/app/modules/defender/defender.service.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    if (response.ok) {
+      const lawyer = await response.json()
+      const lawyerMapped = {
+        ...mapToLawyer(lawyer),
+      }
+      return lawyerMapped
+    }
+
+    const reason = await response.text()
+    console.error('Failed to get lawyer:', reason)
```
Comment (gudjong) [inline apps/judicial-system/api/src/app/modules/defender/defender.service.ts:56]: Please do not use console logging and please do not log as error if you still want to log.

**Review by gudjong — CHANGES_REQUESTED** [review]

**Review by brynjarorng — APPROVED** [review]
LGTM

**Review by gudjong — APPROVED** [review]
Looks great

**Pull request merged**